plugins/experimental/ts_lua/ts_lua_fetch.c

```diff
@@ -253,25 +253,30 @@ ts_lua_fetch_one_item(lua_State *L, const char *url, size_t url_len, ts_lua_fetc
   /* cliaddr */
   memset(&clientaddr, 0, sizeof(clientaddr));
   clientaddr.sin_family = AF_INET;
-  rc                    = 0;
+  const char *p         = NULL;
 
   if (tb) {
     lua_pushlstring(L, "cliaddr", sizeof("cliaddr") - 1);
     lua_gettable(L, -2);
 
     if (lua_isstring(L, -1)) {
       addr = luaL_checklstring(L, -1, &addr_len);
-      rc   = sscanf(addr, "%15s:%d", ipstr, &port);
-      if (rc == 2) {
+      p    = strstr(addr, ":");
+      if (p && addr_len < 32) {
+        port = atoi(p + 1);
+        strncpy(ipstr, addr, p - addr);
+        ipstr[p - addr]     = 0;
         clientaddr.sin_port = htons(port);
         inet_aton(ipstr, (struct in_addr *)&clientaddr.sin_addr.s_addr);
```

Comment:
what if inet_aton() is invalid and returns 0? Can we take care of this case as well and use the default (i.e. set p = NULL)?